core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -13321,6 +13323,284 @@ class KafkaApisTest extends Logging {
+        new GetReplicaLogInfoResponseData.TopicPartitionLogInfo()
+          .setTopicId(topic2.topicId())
+          .setPartitionLogInfo(Collections.singletonList(
+            new GetReplicaLogInfoResponseData.PartitionLogInfo()
+              .setErrorCode(Errors.LOG_DIR_NOT_FOUND.code())
+          ))) asJava)
+      .setBrokerEpoch(brokerEpoch)
+
+    verifyGetReplicaLogInfoRequest(builder, { response =>
+      assertEquals(expected, response.data())
+    })
+  }
+
+  @Test
+  def testGetReplicaLogInfoUnknownTopic(): Unit = {
+    val expectedPartition = 1
+    val expectedUuid = Uuid.randomUuid()
+    val builder = new GetReplicaLogInfoRequest.Builder(new GetReplicaLogInfoRequestData()
+      .setTopicPartitions(
+        Collections.singletonList(new GetReplicaLogInfoRequestData.TopicPartitions()
+          .setTopicId(expectedUuid)
+          .setPartitions(Collections.singletonList(expectedPartition)))))
+    metadataCache = mock(classOf[MetadataCache])
+    when(metadataCache.getTopicName(expectedUuid)).thenReturn(Optional.empty())
+
+    val expectedResponseData = new GetReplicaLogInfoResponseData()
+      .setBrokerEpoch(brokerEpoch)
+      .setTopicPartitionLogInfoList(Collections.singletonList(
+        new GetReplicaLogInfoResponseData.TopicPartitionLogInfo()
+          .setTopicId(expectedUuid)
+          .setPartitionLogInfo(Collections.singletonList(
+            new GetReplicaLogInfoResponseData.PartitionLogInfo()
+              .setPartition(expectedPartition)
+              .setErrorCode(Errors.UNKNOWN_TOPIC_ID.code())
+          ))
+      ))
+    verifyGetReplicaLogInfoRequest(builder, { response =>
+      assertEquals(expectedResponseData, response.data())
+    })
+  }
+
+  @Test
+  def testGetReplicaLogInfoRequestTooManyTopics(): Unit = {
+    // 100 topics, 20 partitions per topic = 2k topic-partitions
+    // only first 1000 should be sent back and truncated = true
+    val numberUuids = 100
+    val numberPartitions = 20
+    val uuids: List[Uuid] = (1 to numberUuids).map(_ => Uuid.randomUuid()).toList
+    val tps = uuids.map(new GetReplicaLogInfoRequestData.TopicPartitions()
+      .setTopicId(_)
+      .setPartitions((1 to numberPartitions).map(new Integer(_)).asJava))
+    val builder = new GetReplicaLogInfoRequest.Builder(
+      new GetReplicaLogInfoRequestData().setTopicPartitions(tps asJava))
+    val expectedLogEndOffset = 10L
+    val expectedLeaderEpoch = 2
+    val expectedLatestEpoch = 3
+
+    def mockTopicName(uuid: Uuid, idx: Int): String = s"topic-idx-$idx-with-uuid-$uuid"
+
+    metadataCache = mock(classOf[MetadataCache])
+    uuids.zipWithIndex.foreach { case (uuid, idx) =>
+      when(metadataCache.getTopicName(uuid)).thenReturn(Optional.of(mockTopicName(uuid, idx)))
+    }
+
+    val expectedResponseData =
+      new GetReplicaLogInfoResponseData().setHasMoreData(true).setBrokerEpoch(brokerEpoch)
+    uuids.take(50).zipWithIndex.foreach { case (uuid, idx) =>
```

Comment:
nit, could you highlight that we only include the first 50 topics / 1000 partitions with a comment? I know you already have a comment at the top of the test which says a similar thing, but this would help folks find the logic faster